In [1]:
import pandas as pd
#Read the thanksgiving.csv file. This file isn't encoded normaly so, I specify "Latin-1" for encoding
data = pd.read_csv("thanksgiving.csv",encoding = "Latin-1")
data.head()

RespondentID Do you celebrate Thanksgiving?  \
0    4337954960                            Yes   
1    4337951949                            Yes   
2    4337935621                            Yes   
3    4337933040                            Yes   
4    4337931983                            Yes   

  What is typically the main dish at your Thanksgiving dinner?  \
0                                             Turkey             
1                                             Turkey             
2                                             Turkey             
3                                             Turkey             
4                                           Tofurkey             

  What is typically the main dish at your Thanksgiving dinner? - Other (please specify)  \
0                                                NaN                                      
1                                                NaN                                      
2                                                NaN                                      
3                                                NaN                                      
4                                                NaN                                      

  How is the main dish typically cooked?  \
0                                  Baked   
1                                  Baked   
2                                Roasted   
3                                  Baked   
4                                  Baked   

  How is the main dish typically cooked? - Other (please specify)  \
0                                                NaN                
1                                                NaN                
2                                                NaN                
3                                                NaN                
4                                                NaN                

  What kind of stuffing/dressing do you typically have?  \
0                                        Bread-based      
1                                        Bread-based      
2                                         Rice-based      
3                                        Bread-based      
4                                        Bread-based      

  What kind of stuffing/dressing do you typically have? - Other (please specify)  \
0                                                NaN                               
1                                                NaN                               
2                                                NaN                               
3                                                NaN                               
4                                                NaN                               

  What type of cranberry saucedo you typically have?  \
0                                               None   
1                             Other (please specify)   
2                                           Homemade   
3                                           Homemade   
4                                             Canned   

  What type of cranberry saucedo you typically have? - Other (please specify)  \
0                                                NaN                            
1                    Homemade cranberry gelatin ring                            
2                                                NaN                            
3                                                NaN                            
4                                                NaN                            

          ...          \
0         ...           
1         ...           
2         ...           
3         ...           
4         ...           

  Have you ever tried to meet up with hometown friends on Thanksgiving night?  \
0                                                Yes                            
1                                                 No                            


In [2]:
#Display the column names of the dataframe
data.columns

Index(['RespondentID', 'Do you celebrate Thanksgiving?',
       'What is typically the main dish at your Thanksgiving dinner?',
       'What is typically the main dish at your Thanksgiving dinner? - Other (please specify)',
       'How is the main dish typically cooked?',
       'How is the main dish typically cooked? - Other (please specify)',
       'What kind of stuffing/dressing do you typically have?',
       'What kind of stuffing/dressing do you typically have? - Other (please specify)',
       'What type of cranberry saucedo you typically have?',
       'What type of cranberry saucedo you typically have? - Other (please specify)',
       'Do you typically have gravy?',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts',
       'Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots',
       'Which of these side dishes aretypically served

In [3]:
# Explore the values of the column 'Do you celebrate Thanksgiving'
data['Do you celebrate Thanksgiving?'].value_counts()



Yes    980
No      78
Name: Do you celebrate Thanksgiving?, dtype: int64

In [4]:
# Filter out rows in data where the responses of the questin "Do you celebrate Thanksgiving?" is No
data= data[data['Do you celebrate Thanksgiving?']== 'Yes']

In [5]:
# Explore the values in the column "What is typically the main dish at your Thanksgiving dinner?"
data['What is typically the main dish at your Thanksgiving dinner?'].value_counts()

Turkey                    859
Other (please specify)     35
Ham/Pork                   29
Tofurkey                   20
Chicken                    12
Roast beef                 11
I don't know                5
Turducken                   3
Name: What is typically the main dish at your Thanksgiving dinner?, dtype: int64

In [6]:
# Display the column 'Do you typically have gravy?' where the dish in the column 'What is typically the main dish at your Thanksgiving dinner' is Tofurkey
data[data['What is typically the main dish at your Thanksgiving dinner?']== "Tofurkey"]['Do you typically have gravy?']

4      Yes
33     Yes
69      No
72      No
77     Yes
145    Yes
175    Yes
218     No
243    Yes
275     No
393    Yes
399    Yes
571    Yes
594    Yes
628     No
774     No
820     No
837    Yes
860     No
953    Yes
Name: Do you typically have gravy?, dtype: object

In [7]:
# Figure out how many people eat Pies, and what Pies people eat
apple_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple'])
pumpkin_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin'])
pecan_isnull = pd.isnull(data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan'])

ate_pies = (apple_isnull) & (pumpkin_isnull) & (pecan_isnull)
ate_pies.value_counts()


False    876
True     104
dtype: int64

The ate_pies series display False in 876 values. Therefore this is the number of people that ate pies in Thanksgiving day.

In [8]:
# In this step, I generate the function convert_age to extract the first item of the column Age. Example: In the raw Database column Age = 18 - 29, the value will be 18. 
# Also this function remove the '+' character from 60+ and convert the column Age to numeric(int). Finally, it will be generated the column data['int_age'].
def convert_age(row):
    age = row['Age']
    age_is_null = pd.isnull(age)
    if age_is_null == True:
        return None
    age_split = age.split( )[0]
    age_remove = age_split.rstrip("+")
    age_int = int(age_remove)
    return age_int
data['int_age'] = data.apply(convert_age,axis=1)


    

In [9]:
data['int_age'].describe()

/dataquest/system/env/python3/lib/python3.4/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count    947.000000
mean      40.089757
std       15.352014
min       18.000000
25%             NaN
50%             NaN
75%             NaN
max       60.000000
Name: int_age, dtype: float64

This metodology has the disadvantage that take the first number of the interval. If the age of the respondent are between 30 - 44, the age will be 30, and if the respondent is older than 60, the function assign a age of 60. 

I think that is important to take into account the withdraw of this method because assign people less age than they have. 

In [10]:
# Define the function convert_income to generate the column int_income, that cleaned up the column 'How much total combined money did all members of your HOUSEHOLD earn last year?'

def convert_income(row):
    income = row['How much total combined money did all members of your HOUSEHOLD earn last year?']
    income_null = pd.isnull(income)
    if income_null == True:
        return None
    income_split = income.split( )[0]
    if income_split == 'Prefer':
        return None
    income_clean = income_split.replace("$","")
    income_clean_all = income_clean.replace(",","")
    income_int = int(income_clean_all)
    return income_int
data['int_income'] = data.apply(convert_income,axis=1)

In [11]:
data['int_income'].describe()

/dataquest/system/env/python3/lib/python3.4/site-packages/numpy/lib/function_base.py:3834: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


count       829.000000
mean      75965.018094
std       59068.636748
min           0.000000
25%                NaN
50%                NaN
75%                NaN
max      200000.000000
Name: int_income, dtype: float64

The results of the int_income function, take the first item of the category in the value of the column 'How much total combined money did all members of your HOUSEHOLD earn last year?'. 
This could introduce a downward bias in the result, because the methodology assign to the respondent, the lower value of the range category. The people that are within the first range($0 to $9,999) will have an income of 0. 


In [12]:
# Observe if there are correlation between the variables Travel Distance and Income in the Thanksgiving dinner.
# First, I analyse the people who earn under 50000 and the correlation with the travel column
data[data['int_income'] < 50000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         106
Thanksgiving is local--it will take place in the town I live in                      92
Thanksgiving is out of town but not too far--it's a drive of a few hours or less     64
Thanksgiving is out of town and far away--I have to drive several hours or fly       16
Name: How far will you travel for Thanksgiving?, dtype: int64

In [13]:
data[data['int_income'] < 150000]['How far will you travel for Thanksgiving?'].value_counts()

Thanksgiving is happening at my home--I won't travel at all                         281
Thanksgiving is local--it will take place in the town I live in                     203
Thanksgiving is out of town but not too far--it's a drive of a few hours or less    150
Thanksgiving is out of town and far away--I have to drive several hours or fly       55
Name: How far will you travel for Thanksgiving?, dtype: int64

The results show that people who earn more than 150000 are more expected to celebrate Thanksgiving day at their home or in the town they live. 

In [14]:
# In the next two steps, I link friendship with age and income. For this I make 2 pivot tables to analyze these two variables with the Friendsgiving event.
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values = 'int_age')

Have you ever attended a "Friendsgiving?"                  No        Yes
Have you ever tried to meet up with hometown fr...                      
No                                                  42.283702  37.010526
Yes                                                 41.475410  33.976744

In [15]:
data.pivot_table(index='Have you ever tried to meet up with hometown friends on Thanksgiving night?',columns='Have you ever attended a "Friendsgiving?"',values = 'int_income')

Have you ever attended a "Friendsgiving?"                     No           Yes
Have you ever tried to meet up with hometown fr...                            
No                                                  78914.549654  72894.736842
Yes                                                 78750.000000  66019.736842

The pivot table that relate the age with the Friendsgiving event, show that young people are more likely to celebrate the event with friends. The average people who respond 'Yes' in both attending a friendsgiving or tried to meet up with hometown friends on Thanksgiving night have an age of 34 years, while the people that respond 'NO' in both events have an age of 42. 

The second table, show that pepole with less income in average are more expected to celebrate Friendgiving, and to meet up with hometown friend on Thanksgiving night. These two questions are related with friendship. 

We can conclude that friendship, is most common between young people and people with less income in average. 

In [16]:
# In the next steps I will explore more in depth the Database and find out patterns.

# To figure out the most common dessert people eat, I joined all the columns that ask about the dessert people eat in th Thanksgiving dinner with the concat method,
# and then utilize the value_counts method to the whole series.
apples = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Apple cobbler']
blondies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Blondies']
brownies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Brownies']
carrot_cake = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Carrot cake']
cheese_cake = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake']
cookies = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cookies']
fudge = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Fudge']
ice_cream = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream']
peach_cobbler = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Peach cobbler']
none = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None']
other = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify)']
other_1 = data['Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Other (please specify).1']
dessert = pd.concat([apples,blondies,brownies,carrot_cake,cheese_cake,cookies,fudge,ice_cream,peach_cobbler,none,other,other_1],axis=0)
dessert.value_counts()

None                                                           295
Ice cream                                                      266
Cookies                                                        204
Cheesecake                                                     191
Other (please specify)                                         134
Brownies                                                       128
Apple cobbler                                                  110
Peach cobbler                                                  103
Carrot cake                                                     72
Fudge                                                           43
Blondies                                                        16
pie                                                             13
Pie                                                             12
pies                                                             6
Pumpkin pie                                                   

In [18]:
# An interesting point in this project, is to figure out the most common complete meal people eat.
# A way to perform this is to count all the items that compose the complete meal and see the results.
# These items are main dish, side dishes, pie and dessert. For the main dish I just got the result, as well as the dessert.
# Then I will perform this task to side dishes and pie
# Value counts of side dishes
Brussel_sprouts = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Brussel sprouts']
Carrots = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots']
Cauliflower = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Cauliflower']
Corn = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn']
Cornbread = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Cornbread']
Fruit_salad = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Fruit salad']
Green_beans_green_bean_casserole = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole']
Macaroni_cheese = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Macaroni and cheese']
Mashed_potatoes = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes']
Rolls_biscuits = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits']
Squash = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Squash']
Vegetable_salad = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Vegetable salad']
Yams_sweet_potato_casserole = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole']
Other_side_dish = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify)']
Other_side_dish_1 = data['Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify).1']

side_dishes = pd.concat([Brussel_sprouts,Carrots,Cauliflower,Corn,
Cornbread,Fruit_salad,Green_beans_green_bean_casserole,Macaroni_cheese,
Mashed_potatoes,Rolls_biscuits,Squash,Vegetable_salad,Yams_sweet_potato_casserole,
Other_side_dish, Other_side_dish_1], axis=0)

counts = side_dishes.value_counts()
side_dishes = pd.DataFrame({"side_dishes": counts.index, 'number': counts})
side_dishes

number  \
Mashed potatoes                                        817   
Rolls/biscuits                                         766   
Green beans/green bean casserole                       686   
Yams/sweet potato casserole                            631   
Corn                                                   464   
Carrots                                                242   
Cornbread                                              235   
Fruit salad                                            215   
Vegetable salad                                        209   
Macaroni and cheese                                    206   
Squash                                                 171   
Brussel sprouts                                        155   
Other (please specify)                                 111   
Cauliflower                                             88   
broccoli                                                 7   
rice                                                     4   
collard greens                                           3   
Rice                                                     3   
Turnip greens                                            2   
creamed onions                                           2   
greens                                                   2   
Broccoli                                                 2   
Broccoli casserole                                       2   
Asparagus                                                2   
Brocolli                                                 2   
Kale and Colored Greens                                  1   
Deviled Eggs                                             1   
cheeseball                                               1   
Sherbet                                                  1   
depends                                                  1   
...                                                    ...   
Spinach soufflí©                                         1   
Mashed Cauliflower                                       1   
B52 sweet potatoes                                       1   
baked potato, cranberry                                  1   
Spinach                                                  1   
peas                                                     1   
Creamed onions, broccoli                                 1   
squash soufflet                                          1   
mushrooms, broccoli rabe, turnip                         1   
Dumplings                                                1   
Rice, beans, tamales                                     1   
green peas with pearl onion                              1   
gizzards                                                 1   
Oyster cassarole, home baked bread                       1   
Pearl onions                                             1   
Peas and a jello salad                                   1   
corn green beans                                         1   
Baked scallop potatoes, rice, sushi, Chinese no...       1   
Yorkshire pudding, mixed vegetable, apple sauce...       1   
fresh green peas                                         1   
plain sweet potatoes, brocolli, relish tray, oy...       1   
salad                                                    1   
Antipasta                                                1   
Fish                                                     1   
baked sweet potatoes, broccoli & cheese sauce            1   
Cajun Cressing (Crayfish, gorund beef & rice, e...       1   
wild rice                                                1   
stuffing and creamed spinach                             1   
Asian vinagrette salad                                   1   
Potato salad, wild rice & mushroom casserole             1   

                                                                                          side_dishes  
Mashed potatoes                                                                       Mashed potatoes  

In [ ]:
# Value counts of Pie
Apple = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple']
Buttermilk = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk']
Cherry = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry']
Chocolate = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate']
Coconut_cream = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Coconut cream']
Key_lime = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Key lime']
Peach = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Peach']
Pecan = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan']
Pumpkin = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin']
Sweet_Potato = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato']
No_apply = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - None']
Other = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify)']
Other_1 = data['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Other (please specify).1']

pie = pd.concat([Apple,Buttermilk,Cherry,Chocolate,Coconut_cream,Key_lime,
Peach,Pecan,Pumpkin,Sweet_Potato,No_apply, Other, Other_1],axis=0)
counts = pie.value_counts()
pie = pd.DataFrame({"Pie": counts.index, 'number': counts})
pie

Analyzing the value counts of the items in the meal, it concludes that Mashed Potatoes ,Pumpkin and Ice Creams are the most side dishes , pie and dessert respectevely that people eat for Thanksgiving day. 
Also, there is a lot number of people that don't eat dessert, because the None value for this was 295.

In [19]:
# Make a Pivot Table to find correlation between location and dinner menus
import numpy as np
data.pivot_table(index = 'How would you describe where you live?', values = ['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None'] , aggfunc= 'count')


Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake  \
How would you describe where you live?                                                                                                                       
Rural                                                                                  37                                                                    
Suburban                                                                               94                                                                    
Urban                                                                                  56                                                                    

                                        Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream  \
How would you describe where you live?                                                                                                                      
Rural                                                                                  51                                                                   
Suburban                                                                              142                                                                   
Urban                                                                                  67                                                                   

                                        Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None  \
How would you describe where you live?                                                                                                                 
Rural                                                                                  68                                                              
Suburban                                                                              161                                                              
Urban                                                                                  63                                                              

                                        Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
How would you describe where you live?                                                                                                                        
Rural                                                                                  60                                                                     
Suburban                                                                              126                                                                     
Urban                                                                                  53                                                                     

                                        Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn  \
How would you describe where you live?                                                                                                                     
Rural                                                                                 121                                                                  
Suburban                                                                              236                                                                  
Urban                                                                                 103                                                                  

                                        Which of these side dishes aretypically served at your Thanksgiving dinner? Please se

In relation to pie that people eat in Thanksgiving dinner, the table show that in all the regions the Pumpkin is the pie with the more people eating that pie. Secondly, comes the apple fot the three regions too. 

Respect to the side dishes, the Mashed potates is the side dishes with the more people who eat at Thanksgiving dinner in the Suburban and Urban areas. In the Rural area the side dish that people eat more is Rolls/biscuit.

In terms of the dessert that people eat more in Thanksgiving day, the table show that while in the Rural and Suburban areas the mayority of people don't eat dessert for the Thanksgiving dinner, in the Urban areas most of the people eat Ice Cream. 
As it show before, it's rarely that the mayority of people don't eat dessert for the Thanksgiving dinner.

In [20]:
# Find correlations between age and dinner menus
data.pivot_table(index = 'int_age', values = ['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None'] , aggfunc= 'count')


Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake  \
int_age                                                                                                                       
18.0                                                    39                                                                    
30.0                                                    47                                                                    
45.0                                                    54                                                                    
60.0                                                    47                                                                    

         Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream  \
int_age                                                                                                                      
18.0                                                    73                                                                   
30.0                                                    60                                                                   
45.0                                                    65                                                                   
60.0                                                    62                                                                   

         Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None  \
int_age                                                                                                                 
18.0                                                    31                                                              
30.0                                                    76                                                              
45.0                                                    94                                                              
60.0                                                    90                                                              

         Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
int_age                                                                                                                        
18.0                                                    61                                                                     
30.0                                                    58                                                                     
45.0                                                    70                                                                     
60.0                                                    50                                                                     

         Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn  \
int_age                                                                                                                     
18.0                                                   111                                                                  
30.0                                                   115                                                                  
45.0                                                   137                                                                  
60.0                                                    97                                                                  

         Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole  \
int_age                                                                                  

In respect of the age of the people that take dinner in Thankgiving day, the table show that the Pumpkin is the most pie that people of every age eat in this date. In the second place for all the ages comes the apple pie.

For the side dishes item that people for all the ages eat, the Mashed potates is the side dishe that people in every age eat. Up to the moment, the findings for the age of the people are the same to the findings of the regional areas. The carrots is the item with the less number of people in every age that eat this side dish.

Respect of the desserts in Thanksgiving for the ages, the table show that the mayority of people that have more than 30 years, don't eat dessert. In contrast, the mayority of the people between 18 and 30 eat dessert and 73 of them eat Ice Cream, and 39 eat Cheesecake. Only 31 of people in this range don't eat dessert. 

In the breakdown of the database by age, this table show a different pattern about the people who eat dessert. These are vast mayority of young people. 

In [21]:
# Find correlations between gender and dinner menus 
data.pivot_table(index = 'What is your gender?', values = ['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None'] , aggfunc= 'count')


Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake  \
What is your gender?                                                                                                                       
Female                                                              106                                                                    
Male                                                                 81                                                                    

                      Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream  \
What is your gender?                                                                                                                      
Female                                                              132                                                                   
Male                                                                128                                                                   

                      Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None  \
What is your gender?                                                                                                                 
Female                                                              155                                                              
Male                                                                136                                                              

                      Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots  \
What is your gender?                                                                                                                        
Female                                                              110                                                                     
Male                                                                129                                                                     

                      Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn  \
What is your gender?                                                                                                                     
Female                                                              220                                                                  
Male                                                                240                                                                  

                      Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole  \
What is your gender?                                                                                                                                                 
Female                                                              369                                                                                              
Male                                                                307                                                                                              

                      Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes  \
What is your gender?                                                                                                                                
Female                                                              439                                                                             
Male                                                                366                                                                  

When I segment the database by gender, the table show that the Pumpkin is the pie that both woman and men eat more. In the second place, come the Apple pie, for both sex. 

Respect of the side dishes, the Mashed potates has the most responses for woman and men.  In the second place it's possible to find Rolls/biscuits side dish for both sex. On the other hand, the carrots has the less responses for both women and men. This pattern is the same for the segmentation of the database by regional zones or age.

Lastly, the vast mayority of female and male don't eat dessert in Thanksgiving dinner. 

In [22]:
# Find correlations between income and dinner menus
data.pivot_table(index = 'int_income', values = ['Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Apple', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Buttermilk','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Cherry', 'Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Chocolate','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pumpkin','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Pecan','Which type of pie is typically served at your Thanksgiving dinner? Please select all that apply. - Sweet Potato','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Mashed potatoes','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Rolls/biscuits','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Green beans/green bean casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Yams/sweet potato casserole','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Corn','Which of these side dishes aretypically served at your Thanksgiving dinner? Please select all that apply. - Carrots','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake','Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None'] , aggfunc= 'count')


Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Cheesecake  \
int_income                                                                                                                       
0.0                                                        11                                                                    
10000.0                                                    15                                                                    
25000.0                                                    40                                                                    
50000.0                                                    20                                                                    
75000.0                                                    24                                                                    
100000.0                                                   23                                                                    
125000.0                                                    6                                                                    
150000.0                                                    4                                                                    
175000.0                                                    5                                                                    
200000.0                                                    8                                                                    

            Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - Ice cream  \
int_income                                                                                                                      
0.0                                                        17                                                                   
10000.0                                                    19                                                                   
25000.0                                                    41                                                                   
50000.0                                                    36                                                                   
75000.0                                                    32                                                                   
100000.0                                                   27                                                                   
125000.0                                                   13                                                                   
150000.0                                                    8                                                                   
175000.0                                                    4                                                                   
200000.0                                                   24                                                                   

            Which of these desserts do you typically have at Thanksgiving dinner? Please select all that apply.   - None  \
int_income                                                                                                                 
0.0                                                        12                                                              
10000.0                                                     9                                                              
25000.0                                                    50                                                              
50000.0                                                    39                                                              
75000.0                                                    45                                                              
100000.0                          

The segmenation by income show that in all categories, the Pumpkin pie is the most pie that people for all categories of income eat. In the second place for all the categories comes the Apple pie.

Respect the side dishes, the Mashed potatoes is the side dish that people for all categories of income eat in Thanksgiving day. As before with the rest of the segmentation, the carrots side dish is the less side dish that people for all income categories eat in the Thanskgiving day. 

In relation of the dessert that people with different income eat in Thanksgiving, the table show that people with less income and people in the highest percentile of income, are more prone to eat dessert in this event. In contrast people who earn between 75000 and 175000 is less willing to eat dessert.

In [23]:
# The last feature, I will cover  in this project is to identify how many people work on Thanksgiving. To determine this I will display the column 
# 'Do you work on retail' that could be Yes or No. 
data['Do you work in retail?'].value_counts()

No     881
Yes     70
Name: Do you work in retail?, dtype: int64

It is reasonable that people who work in retail should work on Thanksgiving day because the following day of Thanksgiving
is Black Friday. The retail sector is the most recipient of BlackFriday and the day before have to make suitable the store and prepare to the next day.

Then, around 70 people of the respondents should work on Thanksgiving day. 